pkg/controllers/awsloadbalancercontroller/deployment_test.go

```diff
@@ -647,3 +711,202 @@ func TestEnsureDeployment(t *testing.T) {
 		})
 	}
 }
+
+func TestHasSecurityContextChanged(t *testing.T) {
+	for _, tc := range []struct {
+		name      string
+		currentSC *corev1.SecurityContext
+		desiredSC *corev1.SecurityContext
+		changed   bool
+	}{
+		{
+			name:      "current RunAsNonRoot is nil",
+			currentSC: &corev1.SecurityContext{},
+			desiredSC: &corev1.SecurityContext{RunAsNonRoot: pointer.BoolPtr(false)},
+			changed:   true,
+		},
+		{
+			name:      "desired RunAsNonRoot is nil",
+			currentSC: &corev1.SecurityContext{RunAsNonRoot: pointer.BoolPtr(false)},
+			desiredSC: &corev1.SecurityContext{},
+			changed:   true,
+		},
+		{
+			name:      "RunAsNonRoot changes true->false",
+			currentSC: &corev1.SecurityContext{RunAsNonRoot: pointer.BoolPtr(true)},
+			desiredSC: &corev1.SecurityContext{RunAsNonRoot: pointer.BoolPtr(false)},
+			changed:   true,
+		},
+		{
+			name:      "RunAsNonRoot changes false->true",
+			currentSC: &corev1.SecurityContext{RunAsNonRoot: pointer.BoolPtr(true)},
+			desiredSC: &corev1.SecurityContext{RunAsNonRoot: pointer.BoolPtr(false)},
+			changed:   true,
+		},
+		{
+			name:      "RunAsNonRoot changes is same",
+			currentSC: &corev1.SecurityContext{RunAsNonRoot: pointer.BoolPtr(true)},
+			desiredSC: &corev1.SecurityContext{RunAsNonRoot: pointer.BoolPtr(true)},
+			changed:   false,
+		},
+		{
+			name:      "current Privileged is nil",
+			currentSC: &corev1.SecurityContext{},
+			desiredSC: &corev1.SecurityContext{Privileged: pointer.BoolPtr(false)},
+			changed:   true,
+		},
+		{
+			name:      "desired Privileged is nil",
+			desiredSC: &corev1.SecurityContext{},
```

Comment:
This test case results in `changed == true` because `RunAsNonRoot` condition precedes the `Priviledged` condition.